components/brave_wallet/browser/ethereum_provider_impl.cc

```diff
@@ -158,7 +158,7 @@ void EthereumProviderImpl::AddEthereumChain(const std::string& json_payload,
                             "", true);
     return;
   }
-  auto chain = brave_wallet::ValueToEthNetworkInfo(*list.begin());
+  auto chain = brave_wallet::ValueToEthNetworkInfo(*list.begin(), true);
```

Comment:
This new filtering perfectly fits separate method for Value -> NetworkInfo conversion I've made for parsing wallet_addEthereumChain payload in a pending PR [here ](https://github.com/brave/brave-core/pull/14104/files#diff-46ae066e01b73c7dfe02c4a519a7a77b8f4a66c4e230ea2301e4fd84cca322a1R98) 
The problem was that brave's and Eip3085 base::Value view of NetworkInfo started to differ very much. 
Now I will have to fix conflicts precisely 🙃